core/src/main/java/org/testcontainers/DockerClientFactory.java

```diff
@@ -39,6 +35,7 @@
 
     private static final List<DockerConfigurationStrategy> CONFIGURATION_STRATEGIES =
```

Comment:
Can we make it extendable? IMO ConfigurationStrategy is a good enough abstraction to make it pluggable, i.e. I see some use cases, plus an ability to work around unsupported new stuff like Docker4Mac, or maybe some brand new way of running Docker?